File: C++/08_기본수학_2/Problem_01.cpp
/* 08단계 [ 기본수학 2 ] */
// 제목 : 소수 찾기
// 문제 번호 : 1978

#include <iostream>
using namespace std;

bool prime_number(int n)
{
    if (n < 2)
        return false;
    else
        for (int i = 2; i * i <= n; i++)
            if (n % i == 0)
                return false;
    return true;
}

int main(void)
{
    int N, n, cnt = 0;
    cin >> N;
    for (int i = 0; i < N; i++)
    {
        cin >> n;
        if (prime_number(n))
            cnt++;
    }
    cout << cnt << endl;
}

File: Python/08_기본수학_2/Problem_01.py
def prime_number(n):
    if n < 2:
        return False
    else:
        i = 2
        while i*i <= n:
            if n % i == 0:
                return False
            i += 1
    return True


cnt = 0
N = int(input())
n = map(int, input().split())

for i in n:
    if prime_number(i):
        cnt += 1
print(cnt)
